src/java/io/bazel/rulesscala/worker/GenericWorker.java

```diff
@@ -1,117 +0,0 @@
-import com.google.devtools.build.lib.worker.WorkerProtocol.WorkResponse;
-import java.io.ByteArrayOutputStream;
-import java.io.IOException;
-import java.io.PrintStream;
-import java.nio.file.Files;
-import java.nio.file.Paths;
-import java.util.Arrays;
-import java.util.List;
-
-public class GenericWorker {
-  protected final Processor processor;
-
-  public GenericWorker(Processor p) {
-    processor = p;
-  }
-
-  protected void setupOutput(PrintStream ps) {
-    System.setOut(ps);
-    System.setErr(ps);
-  }
-
-  // Mostly lifted from bazel
-  private void runPersistentWorker() throws IOException {
-    PrintStream originalStdOut = System.out;
-    PrintStream originalStdErr = System.err;
-
-    while (true) {
-      try {
-        WorkRequest request = WorkRequest.parseDelimitedFrom(System.in);
-        if (request == null) {
-          break;
-        }
-        ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        int exitCode = 0;
-
-        try (PrintStream ps = new PrintStream(baos)) {
-          setupOutput(ps);
-
-          try {
-            processor.processRequest(request.getArgumentsList());
-          } catch (Exception e) {
-            e.printStackTrace();
-            exitCode = 1;
-          }
-        } finally {
-          System.setOut(originalStdOut);
-          System.setErr(originalStdErr);
-        }
-
-        WorkResponse.newBuilder()
-            .setOutput(baos.toString())
-            .setExitCode(exitCode)
-            .build()
-            .writeDelimitedTo(System.out);
-        System.out.flush();
-      } finally {
-        System.gc();
-      }
-    }
-  }
-
-  public static <T> String[] appendToString(String[] init, List<T> rest) {
-    String[] tmp = new String[init.length + rest.size()];
-    System.arraycopy(init, 0, tmp, 0, init.length);
-    int baseIdx = init.length;
-    for (T t : rest) {
-      tmp[baseIdx] = t.toString();
-      baseIdx += 1;
-    }
-    return tmp;
-  }
-
-  public static String[] merge(String[]... arrays) {
-    int totalLength = 0;
-    for (String[] arr : arrays) {
-      totalLength += arr.length;
-    }
-
-    String[] result = new String[totalLength];
-    int offset = 0;
-    for (String[] arr : arrays) {
-      System.arraycopy(arr, 0, result, offset, arr.length);
-      offset += arr.length;
-    }
-    return result;
-  }
-
-  private boolean contains(String[] args, String s) {
-    for (String str : args) {
-      if (str.equals(s)) return true;
-    }
-    return false;
-  }
-
-  private static List<String> normalize(List<String> args) throws IOException {
-    if (args.size() == 1 && args.get(0).startsWith("@")) {
-      return Files.readAllLines(Paths.get(args.get(0).substring(1)), UTF_8);
-    } else {
-      return args;
-    }
-  }
-
-  /** This is expected to be called by a main method */
-  public void run(String[] argArray) throws Exception {
-    if (contains(argArray, "--persistent_worker")) {
```

Comment:
the `contains` allowed one to add extra configs to the persistent worker.
You think that's redundant or worst case we'll add it later on as the additional arguments?

Reply:
I may be mistaken, but I thought Bazel controls all main args when launching the persistent worker, so there is no need to do fancy checking.

Reply:
🤷‍♂ tests pass so let's keep it as you did. Worst case someone will add a FR